api/test/test_covjson.py

```diff
@@ -0,0 +1,104 @@
+
+    # Assert that the coverage collection has the correct parameter
+    # TODO: Change parameter name when parameter names have been decided
+    assert "ff" in coverage_collection.parameters.keys()
+
+    # Check that correct values exist in the coverage collection
+    assert 9.21 in coverage_collection.ranges["ff"].values
+
+    # compare the coverage collection with the compare data
+    # TODO: Modify compare data when parameter names have been decided
+    coverage_collection_json = json.loads(coverage_collection.model_dump_json(exclude_none=True))
+    assert coverage_collection_json == compare_data
+
+
+def test_multiple_parameter_convert():
+    test_data = load_json("test/test_data/test_multiple.json")
+    compare_data = load_json("test/test_data/test_multiple_compare.json")
+
+    response = create_mock_obs_response(test_data)
+
+    coverage_collection = Covjson().convert(response)
+
+    assert coverage_collection is not None
+
+    assert type(coverage_collection) is Coverage
+
+    # Check that the coverage collection has the correct parameters
+    # TODO: Change parameter names when parameter names have been decided
+    assert all([parameter in ["dd", "ff", "rh", "pp", "tn"] for parameter in coverage_collection.parameters.keys()])
+
+    # Check that correct values exist in the coverage collection
+    assert 230.7 in coverage_collection.ranges["dd"].values
+    assert 1000.48 in coverage_collection.ranges["pp"].values
+
+    # TODO: Modify compare data when parameter names have been decided
+    coverage_collection_json = json.loads(coverage_collection.model_dump_json(exclude_none=True))
+    assert coverage_collection_json == compare_data
+
+
+def test_single_parameter_area_convert():
+    test_data = load_json("test/test_data/test_coverages.json")
+    compare_data = load_json("test/test_data/test_coverages_compare.json")
+
+    response = create_mock_obs_response(test_data)
+
+    coverage_collection = Covjson().convert(response)
+
+    assert coverage_collection is not None
+
+    assert type(coverage_collection) is CoverageCollection
+
+    assert len(coverage_collection.coverages) > 1
+
+    assert all([type(coverage) is Coverage for coverage in coverage_collection.coverages])
+
+    # Check that each coverage has the correct parameter
+    # TODO: Change parameter name when parameter names have been decided
+    assert all(["TA_P1D_AVG" in coverage.parameters.keys() for coverage in coverage_collection.coverages])
+
+    # TODO: Modify compare data when parameter names have been decided
+    coverage_collection_json = json.loads(coverage_collection.model_dump_json(exclude_none=True))
+    assert coverage_collection_json == compare_data
+
+
+def test_empty_response_convert():
+    test_data = load_json("test/test_data/test_empty.json")
+    response = create_mock_obs_response(test_data)
+
+    # Expect to get an HTTPException when converting an empty response
+    with pytest.raises(HTTPException):
```

Comment:
Can we test the error code and the part of the error string (body) that identifies the problem?

Reply:
fixed in https://github.com/EURODEO/e-soh/pull/24/commits/ed77ae35c38c90c7444d05fb4c592ca2c18d59cf